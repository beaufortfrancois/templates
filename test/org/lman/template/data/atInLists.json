{
  "foo": [
    {"bar": 10, "inner": [{"bar": 100}, {"bar": 110}]},
    {"bar": 20, "inner": [{"bar": 200}, {"bar": 210}]}
  ]
}
