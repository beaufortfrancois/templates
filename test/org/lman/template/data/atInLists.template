Expanded lists:
{{#foo}}
* This won't render because it's not prefixed by @: {{bar}}.
* This will: {{@.bar}}.
* Partial won't pass on the current scope: {{+partials.one}}.
* Nor this: {{+partials.one arg:bar}}.
* This one will: {{+partials.one arg:@.bar}}.
{{/foo}}

Inline lists:
* Without args: {{#foo +partials.one /}}.
* With non-@ arg: {{#foo +partials.one arg:bar /}}.
* With @ arg: {{#foo +partials.one arg:@.bar /}}.

Inheriting lists:
{{#foo}}
{{#@.inner}}
* This should take its value from the inner list: {{@.bar}}.
{{/@.inner}}
* Templates can explicitly scope @ if they want to access things further up.
{{#@}}
{{#inner}}
* First comes from foo, the other from inner: {{bar}}, {{@.bar}}.
{{/inner}}
{{/@}}
{{/foo}}
