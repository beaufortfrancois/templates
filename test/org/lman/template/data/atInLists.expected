Expanded lists:
* This won't render because it's not prefixed by @: .
* This will: 10.
* Partial won't pass on the current scope: .
* Nor this: .
* This one will: 10.
* This won't render because it's not prefixed by @: .
* This will: 20.
* Partial won't pass on the current scope: .
* Nor this: .
* This one will: 20.

Inline lists:
* Without args: .
* With non-@ arg: .
* With @ arg: 1020.

Inheriting lists:
* This should take its value from the inner list: 100.
* This should take its value from the inner list: 110.
* Templates can explicitly scope @ if they want to access things further up.
* First comes from foo, the other from inner: 10, 100.
* First comes from foo, the other from inner: 10, 110.
* This should take its value from the inner list: 200.
* This should take its value from the inner list: 210.
* Templates can explicitly scope @ if they want to access things further up.
* First comes from foo, the other from inner: 20, 200.
* First comes from foo, the other from inner: 20, 210.
